now/executor/indexer/elastic/elastic_indexer.py

```diff
@@ -396,3 +424,52 @@ def _extract_embeddings(
             print('No embeddings extracted')
             raise
         return embeddings
+
+    @staticmethod
+    def _join_docs_matrix_into_chunks(
+        docs_matrix: List[DocumentArray], on: str = 'index'
+    ) -> DocumentArray:
+        """
+        Transform a matrix of DocumentArray's into one DocumentArray, by adding Documents to the chunk level.
+        If we are joining a docs_matrix during searching, we will keep both the clip text embedding
+        and sbert embedding generated for the text document chunk. Otherwise, if we are joining a docs_matrix
+        during indexing, we will only keep the sbert embedding and the image embedding generated by clip as
+        chunk documents.
+
+        :param on: str = 'index',
+        :param on: the endpoint which is called, if 'index', then we keep sbert embedding
+            and clip image embedding; if 'search', we keep sbert embedding and clip text embedding.
+        """
+        new_da = DocumentArray()
+        if on == 'search':
+            for doc1, doc2 in zip(*docs_matrix):
+                new_doc = Document(text=doc1.chunks[0].text)
+                if (
+                    len(doc1.chunks[0].embedding) == 768
+                    and len(doc2.chunks[0].embedding) == 512
+                ):
+                    new_doc.chunks.extend(doc1.chunks)
+                    new_doc.chunks.extend(doc2.chunks)
+                elif (
+                    len(doc1.chunks[0].embedding) == 512
+                    and len(doc2.chunks[0].embedding) == 768
+                ):
+                    new_doc.chunks.extend(doc2.chunks)
+                    new_doc.chunks.extend(doc1.chunks)
+                else:
+                    raise Exception('Embedding size not supported')
+                new_da.append(new_doc)
+        else:
+            for doc1, doc2 in zip(*docs_matrix):
+                new_doc = Document()
+                text_chunks = [
+                    c for c in doc1.chunks if c.text is not None and c.text != ''
+                ]  # doc1 from SBert with text embedding
+                image_chunks = [
+                    c for c in doc2.chunks if c.uri is not None and c.uri != ''
```

Comment:
I think this could also work?
```suggestion
                    c for c in doc1.chunks if c.text
                ]  # doc1 from SBert with text embedding
                image_chunks = [
                    c for c in doc2.chunks if c.uri
```

Reply:
Done.